File: AddOns/Share/Categories/StopLocation+Shareable.m
//
//  StopLocation+Shareable.m
//  Pods
//
//  Created by Adrian Schoenig on 24/06/2016.
//
//

#import "StopLocation+Shareable.h"

#import "TKShareHelper.h"

@implementation StopLocation (Shareable)

- (NSURL *)shareURL
{
  return [TKShareHelper stopURLForStopCode:self.stopCode
                             inRegionNamed:self.region.name
                                    filter:self.filter];
}

@end

File: AddOns/Share/Categories/StopLocation+Shareable.swift
//
//  StopLocation+Shareable.swift
//  TripKit
//
//  Created by Adrian Schoenig on 30/6/17.
//  Copyright © 2017 SkedGo. All rights reserved.
//

import Foundation

#if TK_NO_FRAMEWORKS
#else
  import TripKit
#endif

extension StopLocation: TKURLShareable {
  
  public var shareURL: URL? {
    guard let regionName = region?.name else { return nil }
    return TKShareHelper.createStopURL(stopCode: stopCode, inRegionNamed: regionName, filter: filter)
  }
  
}
